pkg/tests/tasks/security/certificate/alpn_filter_non_istio_test.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/ns"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+func TestAlpnFilterDisabledForNonIstioTraffic(t *testing.T) {
+	test.NewTest(t).Groups(test.Full).Run(func(t test.TestHelper) {
+
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns.Foo)
+			oc.RecreateNamespace(t, meshNamespace)
+		})
+
+		t.Log("Deploying SMCP")
+		ossm.DeployControlPlane(t)
+
+		// We need the cluster IP of the ingress gateway service to override DNS resolution during the connection test
+		ingressGwServIP := oc.DefaultOC.Invoke(t, `oc get service/istio-ingressgateway -o=jsonpath='{.spec.clusterIP}' -n istio-system`)
+
+		t.Log("Deploying nginx app")
+		app.InstallAndWaitReady(t, app.Nginx(ns.Foo))
+		oc.CreateTLSSecret(t, meshNamespace, "nginx-server-certs", env.GetRootDir()+"/sampleCerts/nginx.example.com/nginx.example.com.key", env.GetRootDir()+"/sampleCerts/nginx.example.com/nginx.example.com.crt")
+
+		templParamSets := []nginx_ingressgateway_yaml_tmpl_params{
+			{
+				Ns:                ns.Foo,
+				Subset:            false,
+				PortLevelSettings: false,
+			},
+			{
+				Ns:                ns.Foo,
+				Subset:            true,
+				PortLevelSettings: false,
+			},
+			{
+				Ns:                ns.Foo,
+				Subset:            false,
+				PortLevelSettings: true,
+			},
+			{
+				Ns:                ns.Foo,
+				Subset:            true,
+				PortLevelSettings: true,
+			},
+		}
+
+		for _, tmplParam := range templParamSets {
+
+			t.NewSubTest(fmt.Sprintf("Testing connection through gateway with parameters: %+v", tmplParam)).Run(func(t test.TestHelper) {
+
+				t.Cleanup(func() {
+					oc.DeleteFromTemplate(t, ns.Foo, nginx_ingressgateway_yaml_tmpl, tmplParam)
+				})
+
+				t.Log(fmt.Sprintf("Applying gateway, virtual service, and destination rule for the nginx app using parameters: %+v", tmplParam))
+				oc.ApplyTemplate(t, ns.Foo, nginx_ingressgateway_yaml_tmpl, tmplParam)
+
+				retry.UntilSuccessWithOptions(t, retry.Options().MaxAttempts(15).DelayBetweenAttempts(time.Second), func(t test.TestHelper) {
+
+					oc.Exec(t,
+						pod.MatchingSelectorFirst("app=istio-ingressgateway,istio=ingressgateway", meshNamespace),
+						"istio-proxy",
+						// We need to connect to the ingress gateway as if its hostname was 'nginx.example.com'
+						fmt.Sprintf(`curl -s -o /dev/null -w '%%{http_code}' https://nginx.example.com/index.html --resolve nginx.example.com:443:%s --insecure`, ingressGwServIP),
+						assert.OutputContains("200", "Connection succeeded.", "Connection failed."))
+				})
+			})
+		}
+	})
+}
+
+type nginx_ingressgateway_yaml_tmpl_params struct {
```

Comment:
nit: camel case is the common code style in Go:
```suggestion
type nginxIngressgatewayYamlTmplParams struct {
```